### ISSUE TITLE
cpr.h includes windows.h on Windows

### ISSUE THREAD

**Issue by cschreib** [libcpr/cpr#700]
On windows, `cpr.h` includes `curl.h`, which itself includes `windows.h`.

## Description
The `windows.h` header is a very annoying header, which defines a number of common words/expressions as macros (such as `min`, `max`, `CopyFile`, etc.). It is usually a bad idea to include it in a header, as it then pollutes all files including said header. For example, it prevents us from creating our own function called `CopyFile`, unless we explicitly `#undef CopyFile` after including `cpr.h`. 

## Expected Behavior
Including `cpr.h` should only include CPR types, and should not define annoying macros.

## Actual Behavior
Including `cpr.h` includes `windows.h` :( Here is one of the include chains I could spot:
 - `cpr/cpr.h` -> `cpr/response.h` -> `curl/curl.h` -> `winsock2.h` -> `windows.h`

## Possible Fix
Including `curl.h` should be avoided in CPR headers.

## Steps to Reproduce
1. Include `cpr.h` on a Windows development environment
2. Watch the macros being defined! (e.g. `CopyFile` -> `CopyFileA`)

## Context
We were including `cpr.h` in one of our core headers, which means our entire code base had to live with `windows.h` being included. We since managed to move the `#include cpr.h` to just the `.cpp` files that needed it, but it required some annoyance (e.g., can't forward-declare `cpr::Header` because it is a typedef).

## Your Environment
* Version used: 1.6.2
* Where did you get it from (e.g. conan, vcpkg, master, ...): github
* Operating System and version: Windows 10 Pro
* Link to a small example:
```c++
void CopyFile() {  /*...*/ }

#include "cpr/cpr.h"

int main() {
    CopyFile(); // compiler error
}
```

**Comment by KingKili** [comment]
I agree that we should not include curl.h in cpr headers. I can work on this after my exams.
Unfortunately, we currently require curl in the headers for version detection. This propably be also achieved using CMake @COM8 what's your take on this?

**Comment by COM8** [comment]
Agree. In theory we could replace it with `#include <curl/curlver.h>`, which only includes [version information](https://github.com/curl/curl/blob/master/include/curl/curlver.h).
